### ISSUE TITLE
Need to reinstall Omnisidebar on Firefox startup to get sidebar functionality

### ISSUE THREAD

**Issue by SpiroC** [Quicksaver/OmniSidebar#108]
I haven't tried this with a clean profile yet, but with my current FF 42.0 and OmniSidebar 1.6.7.1, I don't get bookmarks that "open in sidebar" working or the "open in sidebar" toolbar button working unless I reinstall it.  Opening the "open in sidebar" bookmark, or pressing the "open in sidebar" toolbar button does nothing unless I re-install OmniSidebar.  The "close sidebar" toolbar "X" button also starts working on reinstall. Fortunately it's a no restart install which I do every morning because I like this extension (twin sidebar especially) thanks.  I have lots of addons installed though so I'm wondering if something else overwrites a relevant setting on startup.  I will retest in a clean profile.

**Comment by Quicksaver** [comment]
Do you mean when you right-click a bookmark, there's no longer any "open in sidebar" option to choose from in the menu?

I believe that was removed from Firefox itself if that's what you mean, in which case I do want to re-implement it, but that will depend on #40.

**Comment by SpiroC** [comment]
Thanks fast reply!!  No, I mean, I have bookmarks with the property within the bookmark "open in sidebar" enabled.  You can set it within the bookmark properties just near where you spec the url.  (if you middle click the bookmark it opens in a new tab, but if you left click it normally it opens in the sidebar).  Such a bookmark (as well as the "open in sidebar" toolbar button which is a bidirectional green arrow) doesn't do anything when clicked, however if I reinstall OmniSidebar on FF startup, they both start working again!  Seeing as no-one else has mentioned it I suspect its clashing with another addon. Thanks again. S

**Comment by Quicksaver** [comment]
I'm sorry for the delay in replying.

Ah yes, I always forget about that checkbox. You're probably right in that it's a conflict with another add-on, everything seems to work fine for me right after starting the browser. If you could test with all your other add-ons disabled except for OSB to confirm, I would appreciate it. I won't be able to do anything unless I know which add-on is causing the conflict. :)

**Comment by Quicksaver** [comment]
Without more information I can't do anything. Please re-open if you have anything to add.

**Comment by SpiroC** [comment]
Hi,
Thank you for follow up.  As you suggested I looked into other addons.  
Browser Console advices addon Treestyle Tabs is having the problem in 
file WindowHelper.js:

     10:25:27.912 TypeError: this.browser is undefined
     TSTWH_overrideGlobalFunctions/SidebarUI.show() windowHelper.js:342
     this.webPanels.open() webPanels.jsm:9
     Modules.LOADMODULE/<() webPanels.jsm:43
     PUIU_openNodeIn() changecode.js line 179 > eval:26
     PUIU_openNodeWithEvent() changecode.js line 179 > eval:3
     BEH_onCommand() browser.js:5973
     oncommand() browser.xul:1
     1 windowHelper.js:342:8

I just decided to live with it.  When it really gets annoying in any 
particular browsing session I simply reinstall Omnisidebar (I keep the 
Omnisidebar install page open on a hidden tab group so I can quickly 
flick to it to reinstall.  It wouldn't work if you hadn't made it a 
no-restart installer!), but  the fix only lasts until I restart Firefox.

thanks,

Spiro Conomos

**Comment by Quicksaver** [comment]
Looking at TST's code, I'm almost positive that is unrelated, and potentially harmless. In short, TST is modifying the sidebar's methods, but OSB is initialized after TST and overrides those modifications. But OSB already knows what to do in case TST is enabled so it can tell it what to do anyway, so both should already work fine.

**Comment by SpiroC** [comment]
Hi Luis,

Thanks for the email.  It sounds like TST wasn't really the culprit, its 
just the only extension giving me any information.  I think maybe at the 
moment I noticed both OSB and TST are working at home (that is, my 
sidebar bookmarks are opening correctly in the sidebar after a restart, 
without having to reinstall OSB) but still not working here at my 
workplace.  Actually I thought maybe you'd already found and fixed it 
with v1.6.9 but I can confirm its still behaving the same here at work.  
I have just exported my extensions config (exported the about:support 
page using Scrapbook Plus).  I will take it home and do some config 
comparisons before getting back to you.  Thanks for the v clever addon,

thanks,

Spiro
